src/app/ClusterStateCache.h

```diff
@@ -217,6 +220,64 @@ class AttributeCache : protected ReadClient::Callback
      */
     CHIP_ERROR Get(const ConcreteAttributePath & path, TLV::TLVReader & reader);
 
+    /*
+     * Retrieve the value of an event from the cache given a concrete event path by decoding
+     * it using DataModel::Decode into the in-out argument 'value'.
+     *
+     * For some types of events, the values for the fields in the event is directly backed by the underlying TLV buffer
+     * and has pointers into that buffer. (e.g octet strings, char strings and lists). Unlike its attribute counterpart,
+     * these pointers are stable and will not change until a call to `ClearEventCache` is called.
+     *
+     * The template parameter EventObjectTypeT is generally expected to be a
+     * ClusterName::Events::EventName::DecodableType, but any
+     * object that can be decoded using the DataModel::Decode machinery will work.
+     *
+     * Notable return values:
+     *      - If the provided attribute object's Cluster and Event IDs don't match that of the event in the cache,
+     *        a CHIP_ERROR_SCHEMA_MISMATCH shall be returned.
+     *
+     *      - If event doesn't exist in the cache, CHIP_ERROR_KEY_NOT_FOUND
+     *        shall be returned.
+     */
+
+    template <typename EventObjectTypeT>
+    CHIP_ERROR Get(EventNumber eventNumber, EventObjectTypeT & value)
```

Comment:
So how does the caller decide which event number goes with which decodable type?

Reply:
Will add comments to this effect.